ssl-config/src/main/java/com/palantir/remoting/ssl/pkcs1/Pkcs1Readers.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.ssl.pkcs1;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+public final class Pkcs1Readers {
+
+    private static final ServiceLoader<Pkcs1Reader> PKCS1_READER_LOADER = ServiceLoader.load(Pkcs1Reader.class);
+    private static Pkcs1Reader reader;
```

Comment:
I think you might want to mark this volatile. 

An alternative pattern here would be to make a Supplier and memoize it (Suppliers.memoize(new Supplier)), which should take care of the thread safety bit for you.